Formula/teleconsole.rb

```diff
@@ -0,0 +1,72 @@
+require "language/go"
+
+class Teleconsole < Formula
+  desc "free service to share your terminal session with people you trust"
+  homepage "https://www.teleconsole.com"
+  url "https://github.com/gravitational/teleconsole/archive/0.3.1.tar.gz"
+  sha256 "663307a1dfe4baadf7e1ed9f5b66b1d203bf9696068e9bcd86e535f286e64d59"
+
+  depends_on "go" => :build
+
+  go_resource "github.com/Sirupsen/logrus" do
+    url "https://github.com/Sirupsen/logrus.git",
+      :revision => "d26492970760ca5d33129d2d799e34be5c4782eb"
+  end
+
+  go_resource "github.com/gravitational/trace" do
+    url "https://github.com/gravitational/trace.git",
+      :revision => "6e153c7add15eb07e311f892779fb294373c4cfa"
+  end
+
+  go_resource "github.com/gravitational/teleport" do
+    url "https://github.com/gravitational/teleport.git",
+      :revision => "002b640a16f097e2f834b4ae33c9edfb81d5798c"
+  end
+
+  go_resource "github.com/jonboulle/clockwork" do
+    url "https://github.com/jonboulle/clockwork.git",
+      :revision => "bcac9884e7502bb2b474c0339d889cb981a2f27f"
+  end
+
+  go_resource "golang.org/x/crypto" do
+    url "https://go.googlesource.com/crypto.git",
+      :revision => "9477e0b78b9ac3d0b03822fd95422e2fe07627cd"
+  end
+
+  go_resource "golang.org/x/net" do
+    url "https://go.googlesource.com/net.git",
+      :revision => "55a3084c9119aeb9ba2437d595b0a7e9cb635da9"
+  end
+
+  go_resource "github.com/fatih/color" do
+    url "https://github.com/fatih/color.git",
+      :revision => "bf82308e8c8546dc2b945157173eb8a959ae9505"
+  end
+
+  go_resource "github.com/mattn/go-colorable" do
+    url "https://github.com/mattn/go-colorable.git",
+      :revision => "d228849504861217f796da67fae4f6e347643f15"
+  end
+
+  go_resource "github.com/mattn/go-isatty" do
+    url "https://github.com/mattn/go-isatty.git",
+      :revision => "66b8e73f3f5cda9f96b69efd03dd3d7fc4a5cdb8"
+  end
+
+  patch :p1 do
+    url "https://github.com/gravitational/teleconsole/commit/71f33e7110afa9dd624061a4c6c36341e5249b69.diff"
```

Comment:
Favour `.patch` and the PR `.patch` because it's easier to strip the `.patch` and see more context. Additionally, needs a comment above noting what the patch is needed for and that it was merged upstream.